lib/internal/Magento/Framework/App/Cache/FlushCacheByTags.php

```diff
@@ -56,36 +50,32 @@ public function __construct(
     }
 
     /**
-     * Clean cache on save object
+     * Clean cache when object is saved
      *
-     * @param AbstractResource $subject
-     * @param \Closure $proceed
-     * @param AbstractModel $object
-     * @return AbstractResource
      * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
-    public function aroundSave(AbstractResource $subject, \Closure $proceed, AbstractModel $object): AbstractResource
-    {
-        $result = $proceed($object);
+    public function afterSave(
+        AbstractResource $subject,
+        AbstractResource $result,
+        AbstractModel $object
+    ): AbstractResource {
         $tags = $this->tagResolver->getTags($object);
         $this->cleanCacheByTags($tags);
 
         return $result;
     }
 
     /**
-     * Clean cache on delete object
+     * Clean cache when object is deleted
      *
-     * @param AbstractResource $subject
```

Comment:
Do not delete the params.

Reply:
The parameters are redundant and duplicative.

Code style guide says that they're unnecessary and should not be here:

1. https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#general-principles

> Include all necessary information without duplication.

2. https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#functions-methods

> The declaration of all arguments (if any) using the @param tag, unless the argument type is indicated in the method signature.

Reply:
Try to run phpdoc on this and see. 

The whole reason for the docblock is to make life easier and code more readable and to make note of what the method does. That link in the standards above is pointing you to the [docblock description](https://docs.phpdoc.org/latest/references/phpdoc/basic-syntax.html#what-is-a-docblock). It's important to keep the doc blocks there to continue support for all IDEs. Not everyone is developing on the same IDE.

Maybe other can weight in on this. But IMO it's not good practice to start removing doc blocks just because PHP now supports strict types.

I would actually change the params on multiple lines instead to respect the PSR-2 line length recommendation instead.

Reply:
The whole reason for the docblock is to make life easier and code more readable and to make note of what the method does. That link in the standards above is pointing you to the [docblock description](https://docs.phpdoc.org/latest/references/phpdoc/basic-syntax.html#what-is-a-docblock). It's important to keep the doc blocks there to continue support for all IDEs. Not everyone is developing on the same IDE.

Maybe other can weight in on this. But IMO it's not good practice to start removing doc blocks just because PHP now supports strict types.

I would actually change the params on multiple lines instead to respect the PSR-2 line length recommendation instead.